jmh-benchmarks/src/main/java/org/apache/kafka/jmh/group_coordinator/ClientSideAssignorBenchmark.java

```diff
@@ -0,0 +1,198 @@
+package org.apache.kafka.jmh.group_coordinator;
+
+import org.apache.kafka.clients.consumer.ConsumerPartitionAssignor;
+import org.apache.kafka.clients.consumer.CooperativeStickyAssignor;
+import org.apache.kafka.clients.consumer.RangeAssignor;
+import org.apache.kafka.clients.consumer.internals.AbstractPartitionAssignor;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.Integer.max;
+import static org.apache.kafka.clients.consumer.internals.AbstractStickyAssignor.DEFAULT_GENERATION;
+
+@State(Scope.Benchmark)
+@Fork(value = 1)
+@Warmup(iterations = 5)
+@Measurement(iterations = 5)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class ClientSideAssignorBenchmark {
+
+    @Param({"10", "50", "100"})
+    private int partitionsPerTopicCount;
+
+    @Param({"100"})
+    private int topicCount;
+
+    @Param({"500", "1000", "10000"})
+    private int memberCount;
+
+    @Param({"true", "false"})
+    private boolean isRackAware;
+
+    @Param({"true", "false"})
+    private boolean isSubscriptionUniform;
+
+    @Param({"true", "false"})
+    private boolean isRangeAssignor;
+
+    @Param({"true", "false"})
+    private boolean isReassignment;
+
+    private Map<String, ConsumerPartitionAssignor.Subscription> subscriptions = new HashMap<>();
+
+    private final int numBrokerRacks = 3;
+
+    private final int replicationFactor = 2;
+
+    protected AbstractPartitionAssignor assignor;
```

Comment:
Hmm makes sense, I guess then it would be more accurate since it would include the time taken to create the metadata to pass to the assignor in the assign function, I'll look into it

Reply:
And we use the PartitionAssignor interface in the server side benchmark